### PR TITLE
parseutils: Updated documentation

### PR DESCRIPTION
**Pull request by ThomasTJdev**
Updated documentation for parseutils module.
- Intro code samples
- See also links
- Runnable examples

refs #10330

### CODE REVIEW

**Comment by narimiran** [comment]
I cannot do a review on the lines you didn't touch, so I need to do it this way:

Can you change current `.. code-block` into `runnableExamples` for `parseHex` and `parseSaturatedNatural`?

File: lib/pure/parseutils.nim [hunk 1]
```diff
@@ -7,9 +7,42 @@
...
+##    let userInput2 = "3 years back"
+##
+##    let startYear = input1[0..skipWhile(input1, Digits)-1] # 2019
+##    let yearsBack = input2[0..skipWhile(input2, Digits)-1] # 3
+##
+##    echo "Examination is in " & $(parseInt(startYear) + parseInt(yearsBack))
+##
+##
+## **See also:**
+## * `pure parsers<lib.html#pure-libraries-parsers>`_ for other parsers
```
Comment (narimiran) [inline lib/pure/parseutils.nim]: They are part of `pure` libraries, but I wouldn't call them "pure parsers".

Just like you mentioned `parsexml` and `json`, I would also mention `parseopt`, `parsecsv` and `parsecfg`, and then in the last place have `other parsers` with this link.

File: lib/pure/parseutils.nim [hunk 2]
```diff
@@ -7,9 +7,42 @@
 #    distribution, for details about the copyright.
 #
 
-## This module contains helpers for parsing tokens, numbers, identifiers, etc.
+## This module contains helpers for parsing tokens, numbers, integers, floats, identifiers, etc.
```
[the remark this answers is not in the document]
Comment (ThomasTJdev) [reply lib/pure/parseutils.nim]: Splitted in two

File: lib/pure/parseutils.nim (hunk 1 above)
Comment (ThomasTJdev) [reply lib/pure/parseutils.nim]: Changed according to review and sorted parse* modules in chronological order.

**Pull request merged**